api-generation/src/main/scala/ai/h2o/sparkling/api/generation/common/OutputResolver.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.api.generation.common
+
+import water.api.API
+
+trait OutputResolver {
+  def resolveOutputs(outputSubstitutionContext: ModelOutputSubstitutionContext): Seq[Parameter] = {
+    val h2oSchemaClass = outputSubstitutionContext.h2oSchemaClass
+
+    val outputs =
+      for (field <- h2oSchemaClass.getFields
```

Comment:
I think in this case it would look better without the for comprehension
so just filter not null -> map field to parameter, but that's probably just a matter of personal preference